src/runtime/types.d.ts

```diff
@@ -32,30 +32,34 @@ export interface ParsedContentInternalMeta {
   /**
    * File type of the content, i.e `markdown`
    */
-  _type?: string
+  _type?: 'markdown' | 'yaml' | 'json' | 'csv'
   /**
    * Path to the file relative to the content directory
    */
   _file?: string
   /**
    * Extension of the file
    */
-  _extension?: string
+  _extension?: 'md' | 'yaml' | 'yml' | 'json' | 'json5' | 'csv'
 }
 
 export interface ParsedContentMeta extends ParsedContentInternalMeta {
   [key: string]: any
 }
 
 export interface ParsedContent extends ParsedContentMeta {
+  /**
+   * Layout
+   */
+  layout: string
```

Comment:
`layout` should be an optional field because it may or may not present in front-matter.
I'm also thinking of moving it to `ParsedContentMeta`

Reply:
Yes, good catch for the undefined.

Should we moved it to ParsedMeta, I don't know. I myself asked the question and I understand that ParsedMeta are calculated and because layout is not, I added here. But i'm completely open to move it.

Reply:
I think `ParsedMeta` is better place for it